src/abstract-server.ts

```diff
@@ -92,6 +94,10 @@ export abstract class AbstractServer extends EventEmitter {
         return serverArguments || [];
     }
 
+    private resolveServerEnv(serverEnv?: NodeJS.ProcessEnv): NodeJS.ProcessEnv {
+        return serverEnv || nodeProcess.env;
```

Comment:
IMO, this should be a shallow merge, like...

```
return {
  ...nodeProcess.env,
  ...serverEnv
};
```

Reply:
I was thinking about it but was not convinced that it is the best idea. I can however change to it if you prefer.

Reply:
Removing all env variables would be impossible using this solution (unless client list each of them).

Reply:
do you want to remove them all?

Reply:
yes